test/simplify_algebra_test.cpp

```diff
@@ -4195,6 +4236,61 @@ TEST_CASE(dot_slice_batch_dims)
     EXPECT(m1.sort() == m2.sort());
 }
 
+
+TEST_CASE(complex_graph_operations)
```

Comment:
What is this test case for? Needs a more descriptive name and probably a docstring. I think it's to prevent the mul from `x_27` is not moved? The `EXPECT` case doesn't seem to make sense for that test.